### PR TITLE
WIP TC-632 Add React Testing Library to atoms

### PR DESCRIPTION
**Pull request by kscoulter**
Adds react testing library tests to editableTotal and modal components.

### CODE REVIEW

File: solution/ui/package.json [hunk 1]
```diff
@@ -31,6 +31,10 @@
   "devDependencies": {
     "@aws-ee/serverless-settings-helper": "workspace:^1.0.0",
     "@aws-ee/serverless-ui-tools": "workspace:^1.0.0",
+    "@testing-library/dom": "^7.29.0",
+    "@testing-library/jest-dom": "^5.11.6",
+    "@testing-library/react": "^11.2.2",
+    "@testing-library/user-event": "^12.6.0",
```
Comment (peggles2) [inline solution/ui/package.json:37]: master merged with this? I think master already has this stuff

File: solution/ui/src/__mocks__/gatsby.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    site: {
+      siteMetadata: {
+        title: `BEP Calc`,
+        description: `test`,
+        author: `SBA`,
+        siteUrl: `test`,
+      },
+    },
+})
+}
```
Comment (peggles2) [inline solution/ui/src/__mocks__/gatsby.js:43]: Nice job with the mocks!!

File: solution/ui/src/components/atoms/editableTotal/editableTotal.test.js [hunk 3]
```diff
@@ -11,77 +12,80 @@ describe('EditableTotal', () => {
...
   }
 
-  const mockEvent = {
-    preventDefault: jest.fn()
-  }
-
-  beforeEach(() => {
-    jest.clearAllMocks()
+  test('displays a correct title', () => {
+    render(<EditableTotal {...sampleProps}/>)
```
Comment (peggles2) [inline solution/ui/src/components/atoms/editableTotal/editableTotal.test.js:16]: These are rendered again over and over. Might be better to put this in a beforeEach

File: solution/ui/src/components/molecules/breakEvenProfileCard/breakEvenProfileCard.test.js [hunk 4]
```diff
@@ -1,20 +1,51 @@
...
+
+  test('it closes confirmation modal after a timeout', async() => {
+    render(<BreakEvenProfileCard {...sampleProps}/>)
+    userEvent.click(screen.queryAllByRole('button')[0])
+    userEvent.click(screen.getByLabelText('apply'))
+    expect(screen.queryByRole('alert')).toBeInTheDocument()
+    await waitFor(() => {
+      expect(screen.queryByRole('alert')).not.toBeInTheDocument()
+    }, {timeout: 4000})
+  })
```
Comment (peggles2) [inline solution/ui/src/components/molecules/breakEvenProfileCard/breakEvenProfileCard.test.js:45]: Nice job with the interaction tests

File: solution/ui/src/pages/index.test.js [hunk 5]
```diff
@@ -1,8 +1,30 @@
...
+  userEvent.click(screen.getByLabelText(totalFixedCostYesRadioText))
+  userEvent.type(screen.getByLabelText('total fixed cost'), '1000')
+  fireEvent.submit(screen.getByTestId('fixedCosts-form'))
+  // test eaditable total onEdit implementation
+  userEvent.click(screen.getByLabelText(`edit Total fixed cost`))
+  userEvent.type(screen.getByRole('input'), '0')
+  expect(screen.getByRole('input')).toHaveValue(10000)
+  userEvent.click(screen.getByLabelText('apply'))
+  expect(screen.getByTestId('value')).toHaveTextContent('10,000')
+})
```
Comment (peggles2) [inline solution/ui/src/pages/index.test.js:26]: For now lets keep the tests we had in the same structure and later lets think if we want to re-organize it.

**Review by peggles2 — CHANGES_REQUESTED** [review]
Looks good nice job with the user interaction tests. Lets discuss later about how/if we want to restructure the tests. Let's keep it the way it is for now.

File: solution/ui/src/components/atoms/editableTotal/editableTotal.test.js (hunk 3 above)
Comment (chris-poe) [reply solution/ui/src/components/atoms/editableTotal/editableTotal.test.js:17]: Is this considered resolved?

Comment (kscoulter) [reply solution/ui/src/components/atoms/editableTotal/editableTotal.test.js:17]: yes

**Review by chris-poe — APPROVED** [review]

**Pull request merged**